### ISSUE TITLE
App crashing when navigating to new screen

### ISSUE THREAD

**Issue by darriuk** [LonelyCpp/react-native-youtube-iframe#110]
**Description**
The YoutubePlayer causes a crash immediately on navigating to another screen, and also sometimes when navigating to the screen that contains the component.

**Error**
No errors in the console or reported on the device by the OS. The app simply closes silently.

**Testing environment**
Device: real 64 bit Nokia device
OS: Android 10
React-native: 0.63.3

**Comment by darriuk** [comment]
OK I have a fix for this.

It's a [crazy bug](https://github.com/react-native-webview/react-native-webview/issues/811) in the WebView library where the fix is to set its opacity to 0.99.

This completely fixed it for me:

```
<YoutubePlayer 
...
    webViewStyle={ {opacity:0.99} }
/>
```

**Comment by LonelyCpp** [comment]
yep, it is related to webview opacity and animations. 

During a screen transition, react-navigation animates the opacity of the whole screen (including the webview) and promptly crashes it.

Another alternative to your solution is to change or remove the default animation on screens that have webviews in them. 

More info here - https://github.com/LonelyCpp/react-native-youtube-iframe/issues/101

**Comment by chichilatte** [comment]
The 99% opacity trick was working for us. Except on Android 5 (and below) where we were getting a white space instead of the video player. We needed to add...
```
import {Platform} from 'react-native';
...
<YoutubePlayer
    webViewProps = {
        {
             androidLayerType:  Platform.OS==="android" && Platform.Version<=22 ? "hardware" : "none" 
        }
    }
...
/>
```
More info on androidLayerType [here](https://github.com/react-native-webview/react-native-webview/blob/master/docs/Reference.md#androidLayerType). We're still unsure whether we should use "hardware" or "software", since some webview hardware rendering has given us crashes in the past.

Our setup: RN 0.63.3, react-native-youtube-iframe 1.4.1

**Comment by 201724468** [comment]
> OK I have a fix for this.
> 
> It's a [crazy bug](https://github.com/react-native-webview/react-native-webview/issues/811) in the WebView library where the fix is to set its opacity to 0.99.
> 
> This completely fixed it for me:
> 
> ```
> <YoutubePlayer 
> ...
>     webViewStyle={ {opacity:0.99} }
> />
> ```

This actually works for my project.
Thx for solving my problem

**Comment by gustavohcastro** [comment]
> OK I have a fix for this.
> 
> It's a [crazy bug](https://github.com/react-native-webview/react-native-webview/issues/811) in the WebView library where the fix is to set its opacity to 0.99.
> 
> This completely fixed it for me:
> 
> ```
> <YoutubePlayer 
> ...
>     webViewStyle={ {opacity:0.99} }
> />
> ```

This bug are insane, how its possible, lol

Thanks for the solution, works for my project too!!!

**Comment by philosophicalfly** [comment]
No F way opacity fixes it

**Comment by dybdeskarphet** [comment]
Why the f does opacity fix this?

**Comment by murrayee** [comment]
> OK I have a fix for this.
> 
> It's a [crazy bug](https://github.com/react-native-webview/react-native-webview/issues/811) in the WebView library where the fix is to set its opacity to 0.99.
> 
> This completely fixed it for me:
> 
> ```
> <YoutubePlayer 
> ...
>     webViewStyle={ {opacity:0.99} }
> />
> ```

niubility
